### PR TITLE
[#795] Include sapling-spend file into bundle for tests

### LINKED ISSUE #795: Include sapling-spend file into bundle for tests
Tests are currently fetching `sapling-spend.params` file and `sapling-output.params` file. So when `download.z.cash` host is down our tests don't work. These files should be already pre-downloaded for tests.

### PR DESCRIPTION
**Pull request by Chlup**
Closes #795

- Added structure `SaplingParamsSourceURL` which holds source URLs for sapling-spend file and sapling-output file. URLs are ultimately passed to `SaplingParameterDownloader`.
- This is done so it's possible to pass different URLs when running tests.
- sapling-spend file and sapling-output file are in tests bundle. And when running tests these files are loaded from bundle and not from network.

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [ ] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [ ] Run the app: Did you run the app and try the changes? 
- [ ] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [ ] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/mater/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage befor and after your changes and when possible, leave it in a better place. [Learn More...](../blob/master/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/master/README.md), [LICENSE.md](../blob/master/LICENSE.md), and [Architecture.md](../blob/master/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,10 @@
 # Unreleased
+- [#795] Include sapling-spend file into bundle for tests
+    
+    This is only an internal change and doesn't change the behavior of the SDK. `Initialier`'s
```
Comment (pacu) [inline CHANGELOG.md:4]:
```suggestion
    This is only an internal change and doesn't change the behavior of the SDK. `Initializer`'s
```

File: Sources/ZcashLightClientKit/Block/CompactBlockProcessor.swift [hunk 2]
```diff
@@ -146,6 +146,7 @@ actor CompactBlockProcessor {
         public var dataDb: URL
         public var spendParamsURL: URL
         public var outputParamsURL: URL
+        let saplingParamsSourceURL: SaplingParamsSourceURL
```
Comment (pacu) [inline Sources/ZcashLightClientKit/Block/CompactBlockProcessor.swift:149]: place constants before vars 🙏🙏🙏

File: Sources/ZcashLightClientKit/Initializer.swift [hunk 3]
```diff
@@ -57,6 +57,15 @@ extension Notification.Name {
     static let connectionStatusChanged = Notification.Name("LightWalletServiceConnectivityStatusChanged")
 }
 
+public struct SaplingParamsSourceURL {
```
Comment (pacu) [inline Sources/ZcashLightClientKit/Initializer.swift:60]: Could you document what this is for and how it should be used?

File: Sources/ZcashLightClientKit/Initializer.swift [hunk 4]
```diff
@@ -80,6 +89,7 @@ public class Initializer {
     private(set) var pendingDbURL: URL
     private(set) var spendParamsURL: URL
     private(set) var outputParamsURL: URL
+    let saplingParamsSourceURL: SaplingParamsSourceURL
```
Comment (pacu) [inline Sources/ZcashLightClientKit/Initializer.swift:92]: please place constant before vars.

**Review by pacu — CHANGES_REQUESTED** [review]

**Review by LukasKorba — APPROVED** [review]

**Review by pacu — APPROVED** [review]
UtACK!

**Pull request merged**